software/src/applets/WTVCO.h

```diff
@@ -0,0 +1,691 @@
+                    break;
+
+                case WAVEFORM_B:
+                    gfxPrint(3, MENU_ROW, "B:");
+                    gfxPrint(wavetable_names[waveform[B]]);
+                    break;
+
+                case WAVEFORM_C:
+                    gfxPrint(3, MENU_ROW, "C:");
+                    gfxPrint(wavetable_names[waveform[C]]);
+                    break;
+
+                default: break;
+            }
+        }
+        gfxLine(0, y+11, 63, y+11);
+        gfxLine(0, 63, 63, 63);
+    }
+
+    void DrawSelector() {
+        if(!EditMode()) {
+            gfxSpicyCursor(1 + (cursor * X_DIV), 23, X_DIV);
+        }
+    }
+
+    void DrawWaveForm() {
+        switch((WTVCO_Cursor)cursor) {
+            case QUANTIZE:
+                // debug menu:
+                // gfxPrint(1, 32, "cv: "); gfxPrint(last_cv);
+                // gfxPrint(1, 32+11, "wt: "); gfxPrint(wt_pos);
+                // break;
+
+                // TODO: add transpose / root note / ref voltage selection
+
+            case WAVEFORM_OUT:
+                gfxRenderWave(OUT);
+                break;
+
+            case WAVEFORM_A:
+                gfxRenderWave(A);
+                break;
+
+            case WAVEFORM_B:
+                gfxRenderWave(B);
+                break;
+
+            case WAVEFORM_C:
+                gfxRenderWave(C);
+                break;
+
+            default: break;
+        }
+    }
+
+
+// WAVETABLE STUFF:
+
+    std::vector<int8_t> InterpolateWaveTable(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            waveform[i] = (int8_t) ((((wt_pos <= 127) * ((127 - wt_pos) * (int)wt[A][i] + wt_pos * (int)wt[B][i]))
+                                   + ((wt_pos > 127) * ((255 - wt_pos) * (int)wt[B][i] + (wt_pos - 128) * (int)wt[C][i]))) / 127);
+        }
+        return waveform;
+    }
+
+    void GenerateWaveTable(int w) {
+        switch(waveform[w]) {
+            case WAVE_SINE:
+                wt[w] = GenerateWaveForm_Sine(WT_SIZE);
+                break;
+            case WAVE_TRIANGLE:
+                wt[w] = GenerateWaveForm_Triangle(WT_SIZE);
+                break;
+            case WAVE_SQUARE:
+                wt[w] = GenerateWaveForm_Square(WT_SIZE);
+                break;
+            case WAVE_SAW:
+                wt[w] = GenerateWaveForm_Sawtooth(WT_SIZE);
+                break;
+            case WAVE_RAMP:
+                wt[w] = GenerateWaveForm_Ramp(WT_SIZE);
+                break;
+
+            default: break;
+        }
+        wt[OUT] = InterpolateWaveTable(WT_SIZE);
+    }
+
+
+// standard waves
+    std::vector<int8_t> GenerateWaveForm_Sine(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            waveform[i] = static_cast<int8_t>(127 * std::sin(TWO_PI * t));
```

Comment:
I wouldn't use `std::sin` because of code size. You can do `arm_sin_q15` or `arm_sin_q31` instead.

Something like:
```
q15_t t = static_cast<q15_t>(i * 32767 / size);
waveform[i] = arm_sin_q15(t) >> 8;
```